packages/react-ui-testing/SeleniumTesting/Controls/Input.cs

```diff
@@ -22,19 +24,83 @@ public void AppendText(string keys)
+        /// </summary>
+        /// <param name="text">Вводимая строка</param>
         public void ClearAndInputText([NotNull] string text)
         {
             ExecuteAction(
                 x =>
                     {
                         Clear();
-                        if(text != "")
+                        if (text == "")
                         {
-                            var inputElement = x.FindElement(By.CssSelector("input"));
-                            inputElement.SendKeys(text);
+                            return;
                         }
+                        var inputElement = x.FindElement(By.CssSelector("input"));
+                        inputElement.SendKeys(text);
                     },
-                $"InputText({text})");
+                $"ClearAndInputText({text})");
+        }
+
+        /// <summary>
+        /// Ввод с фиксированной задержкой после каждого символа.
+        /// Полезно, если фронт не успевает обрабатывать ввод через ClearAndInputText
+        /// </summary>
+        /// <param name="text">Вводимая строка</param>
+        /// <param name="delayMilliseconds">Задержка после ввода символа</param>
+        public void ClearAndInputTextWithDelays([NotNull] string text, int delayMilliseconds = 50)
+        {
+            ExecuteAction(
+                x =>
+                {
+                    Clear();
+                    if (text == "")
+                    {
+                        return;
+                    }
+                    var inputElement = x.FindElement(By.CssSelector("input"));
+                    foreach (var c in text)
+                    {
+                        inputElement.SendKeys(c.ToString());
+                        Thread.Sleep(delayMilliseconds);
+                    }
+
+                },
+                $"ClearAndInputTextWithDelays({text}, {delayMilliseconds})");
+        }
+
+
+        /// <summary>
+        /// Явно дожидается успешного ввода каждого символа.
+        /// Полезно, если фронт не успевает обрабатывать ввод через ClearAndInputText
+        /// </summary>
+        /// <param name="text">Вводимая строка</param>
+        /// <param name="timings">Тайминги для ожидания подстроки</param>
+        public void ClearAndInputTextWaitingEveryChar([NotNull] string text, Timings timings = null)
+        {
+            ExecuteAction(
+                x =>
+                {
+                    Clear();
+                    if (text == "")
+                    {
+                        return;
+                    }
+                    var inputElement = x.FindElement(By.CssSelector("input"));
+                    var buffer = "";
+                    foreach (var c in text)
+                    {
+                        buffer += c;
+                        inputElement.SendKeys(c.ToString());
+                        var buffer1 = buffer;
+                        Value.Wait().That(y => y.AssertEqualTo(buffer1), timings);
```

Comment:
Можно сделать так?

```suggestion
                        Value.Wait().Equals(buffer)
```

Там вроде для `Wait()` глобально устанавливается таймаут в 2 сек, поэтому можно не задавать дополнительный.

---

И с этим методом метод `ClearAndInputTextWithDelays` кажется ненужным. Типа зачем подбираться тайминги, когда есть метод который сам гарантированно вводит текст побуквенно.

Reply:
Value.Wait().Equals(buffer) - мы так будем сравнивать IValueProvider и строку универсальным методом для сравнения объектов. И возвращать он будет true или false. 

А копию буфера пришлось сделать из-за лямбды - иначе переменная попадет в замыкание. Переименовал в currentBuffer, чтобы не так глупо выглядело как buffer1